assets/js/base/hooks/use-previous.js

```diff
@@ -0,0 +1,15 @@
+import { useRef, useEffect } from 'react';
+
+/**
+ * Use Previous from https://usehooks.com/usePrevious/.
+ * @param {mixed} value
+ */
+export const usePrevious = ( value ) => {
+	const ref = useRef();
+
+	useEffect( () => {
+		ref.current = value;
+	}, [ value ] );
+
+	return ref.current;
+};
```

Comment:
I don't know how to best explain it, but if you want to add a comment please go ahead. I added tests.

Reply:
I'm overthinking it.  If there's a non-primitive mutation happening:

a. it will have no affect on usage of `usePrevious` because of the mutation!
b. the mutation may cause other problems but that's outside of the scope of this work completely.